Cover various Python, Django and DRF versions for tests

ChristianKreuzberger opened the pull request:
Hi!

Is there a reason why the tox tests do not cover various newer versions of Python as well as Django and DRF? [Travis](https://travis-ci.org/ChristianKreuzberger/drf-nested-routers) seems to be fine with it.

auvipy commented:
@alanjds django version less then 1.8 should be dropped and drf version less then 3.1 too. better only 3.3+

auvipy commented on tox.ini:
```diff
@@ -1,7 +1,7 @@
 envlist =
        py27-{flake8,docs},
-       {py27,py33,py34}-django{1.6,1.7,1.8}-drf{2.4,3.0,3.1}
+       {py27,py33,py34,py35}-django{1.6,1.7,1.8,1.9,1.10}-drf{2.4,3.0,3.1,3.3,3.5}
```
1.6 and 1.7 could be dropped along with 2.x drf

ChristianKreuzberger replied:
I'm leaving them in for now, as we are not breaking compatibility for now. It's on the ToDo list though for later.

auvipy commented on tox.ini:
```diff
@@ -11,9 +11,13 @@ deps =
        django1.6: Django==1.6.11
```
1.6 and 1.7 could be removed as well

ChristianKreuzberger replied:
Same as before, we're going to be waiting on removing compatibility.

alanjds commented:
@auvipy: I see that Django <= 1.8 should be dropped because LTS is 1.8, but why DRF 3.0 ?

auvipy commented:
as version 3.0 is already 2 years old

anx-ckreuzberger commented:
If you want to support older releases that's fine, as there might still be people that use the old versions.

As already stated in the other PR, this calls for a new major version.

alanjds commented:
Yeah... I guess its better to stay with 3.0+ for now.

alanjds commented:
Just to "open my heart", lately I had thoughts that this project grew too much to the actual organizational scheme. I really like the simplicity of what it is right now, but starts to seems to me that it needs e.g. a deprecation policy, releases roadmap with at least the expected release dates, this kind of stuff...

But how heavy should it be to help people to help it and know what to expect? If too heavy, will harm more than help, in my opinion.

ChristianKreuzberger commented:
I completely understand your thoughts. I wouldn't worry too much about deprecation policies, other than modifying your [setup.py](https://github.com/alanjds/drf-nested-routers/blob/master/setup.py#L90-L94) with the appropriate library versions and making sure that minor releases do not remove any support for the given dependencies.
(This means that major releases should be allowed to break compatibility with "older" versions of some dependencies)

Also, I'm not 100% sure about this (somebody can surely confirm or deny this), but you can specify framework versions in your setup.py, e.g.:
```
        'Framework :: Django',
        'Framework :: Django :: 1.8',
        'Framework :: Django :: 1.9',
        'Framework :: Django :: 1.10',
```
Despite this you can always specify dependencies (as you already did) and make sure nobody installs a "wrong" version of your component:
```
    install_requires=[
        'djangorestframework>=2.4.3',
        'Django>=1.6',
    ],
```

Anyway, I feel like I should finish my promised work on the Nested URL Resolver and maybe AFTER that we can think about making a major release and switch to newer versions of the libraries? :)

alanjds commented:
@ChristianKreuzberger sounds good for me ;)

ChristianKreuzberger commented:
I've simplified my commit and removed the change in README with travis build status.
This PR is ready to be merged IMHO.
@auvipy I also believe that compatibility with old versions of Python, Django and DRF should be removed soon, though @alanjds and myself came to the conclusion that we should not break compatibility now and make a major release soon.

alanjds commented:
Merged.

I am inclined to deprecate Django < 1.8 and DRF < 3.0 in the next major release.

Latest DRF released tests against Django 1.8+ and Python 2.7 & 3.3+, but I dislike 3.3 and want to bump to 2.7 & 3.4+ at least.

auvipy commented:
@alanjds +1 frome me for 2.7 n 3.4+ only
